spec/services/attribute_asserter_spec.rb

```diff
@@ -327,6 +327,31 @@
         end
       end
 
+      context 'service provider specifies attribute bundle' do
+        before do
+          allow(service_provider.metadata).to receive(:[]).with(:attribute_bundle).
+            and_return(%w[email first_name last_name ssn phone])
+          subject.build
+        end
+
+        context 'the service provider is ial1' do
+          let(:service_provider_ial) { 1 }
+
+          it 'only includes uuid, email, aal, and ial (no ial2 attributes)' do
+            expect(user.asserted_attributes.keys).to eq %i[uuid email aal ial]
+          end
+        end
+
+        context 'the service provider is ial2' do
```

Comment:
**Question:** Are you testing the case where no IAL AuthnContext is passed?

Reply:
Thinking about this further... The tests we need to write (As you pointed out) are SAML specific and do not belong here.

Reply:
Oh yeah, you should only really need to add tests for the `IalContext` service object since that's where the change is.